### ISSUE TITLE
Add CBCS content to the demo app

### ISSUE THREAD

**Issue by kuznetcoff777** [google/shaka-player#2457]
Hi! As i know, apple won a battle for CBC method in CMAF drm encryption.
I prepare cmaf without encription as making hls with fmp4 and create mpd manifest for the same fmp4 cnunks.
Now i need DRM stream. Am i correctly undestand that if i do a FairPlay hls stream with fmp4 and prepare correct CENC header in mpd manifest, then shaka would play that stream correctly?

**Comment by joeyparrish** [comment]
Not all platforms support CBCS, though many do.  For example, both CTR and CBCS should work on desktop Chrome and current versions of Android, but may not work on all Widevine devices.  Older smart TVs and set top boxes and possibly older Android versions (not sure) may only support CTR mode because they haven't been updated.

So unfortunately, we can't call it a "won" battle across the board.  Eventually, as older devices are replaced, there will be more consistency in the ecosystem over time.  If you're only supporting the current generation of devices, you _may_ be able to use CBCS everywhere.

Does this answer your question?

**Comment by kuznetcoff777** [comment]
We will have both new and legacy support (but new one with dvr and better quality for example). I wanted to know also if shaka already support such content to play?

**Comment by joeyparrish** [comment]
Support for encryption schemes (CTR vs CBCS) is not up to the player, but to the browser/platform.  If the browser/platform supports an encryption scheme, it should decrypt and play correctly.  If it doesn't, you could expect either a license failure (ideal, because the CDM knows it can't support it) or a decode failure (because the CDM decrypted the content with the wrong scheme).

In the future, the EME API will provide us with a way to [query the encryption schemes](https://wicg.github.io/encrypted-media-encryption-scheme/) supported by the platform so that we can manage this better.  There is already a [polyfill for that](https://github.com/google/eme-encryption-scheme-polyfill) built into Shaka Player.  We have yet to incorporate the query into our code library, though.  You can expect that some time after v2.6 is released, by which time you should see native support for this API in browsers.

**Comment by kuznetcoff777** [comment]
Ok i got it, i thought that player must be additionally prepared to play cbcs content. Do ypu have some sample for cmaf with cbcs encription?

**Comment by joeyparrish** [comment]
No, but we should be able to produce some without too much trouble.  Shaka Packager and Shaka Streamer both support the production of CBCS Widevine content.

**Comment by joeyparrish** [comment]
I'm converting this into a feature request to add CBCS content to the demo.

**Comment by kuznetcoff777** [comment]
Thank you so much

**Comment by joeyparrish** [comment]
I'm going to keep this open to track the addition of CBCS content.
